pkg/distro/fedora/images.go

```diff
@@ -204,19 +203,9 @@ func osCustomizations(
 			CompressionEnabled: true,
 		}
 
-		var tailoringConfig *oscap.TailoringConfig
-		if oscapConfig.Tailoring != nil {
-			remediationConfig.TailoringPath = filepath.Join(oscap.DataDir, "tailoring.xml")
-			tailoringConfig = &oscap.TailoringConfig{
-				RemediationConfig: remediationConfig,
-				TailoredProfileID: fmt.Sprintf("%s_osbuild_tailoring", oscapConfig.ProfileID),
-				Selected:          oscapConfig.Tailoring.Selected,
-				Unselected:        oscapConfig.Tailoring.Unselected,
-			}
-			// we need to set this after the tailoring config
-			// since the tailoring config needs to know about both
-			// the base profile id and the tailored profile id
-			remediationConfig.ProfileID = tailoringConfig.TailoredProfileID
+		tailoringConfig, err := oscap.NewTailoringConfig(oscapConfig.Tailoring, &remediationConfig)
```

Comment:
Does it make sense to expand the helper function to include the things above this line and have it generate the `remediationConfig` from scratch?  Comparing the code in Fedora and RHEL, they seem identical (but I haven't looked very closely).

I mean putting
```go
		var datastream = oscapConfig.DataStream
		if datastream == "" {
			if imageConfig.DefaultOSCAPDatastream == nil {
				return manifest.OSCustomizations{}, fmt.Errorf("No OSCAP datastream specified and the distro does not have any default set")
			}
			datastream = *imageConfig.DefaultOSCAPDatastream
		}
		remediationConfig := oscap.RemediationConfig{
			Datastream:         datastream,
			ProfileID:          oscapConfig.ProfileID,
			CompressionEnabled: true,
		}
```
into the helper function (and renaming it), and then
```go
tailoringConfig, remediationConfig, err := oscap.NewConfigs(oscapConfig, imageConfig.DefaultOSCAPDatastream)
```

Or maybe just moving the initial creation of the `remediationConfig` in the helper and having:
```go
tailoringConfig, remediationConfig, err := oscap.NewConfigs(oscapConfig, datastream)
```
whichever makes more sense.

Reply:
Yeah solid shout. Kinda felt weird only doing this for tailoring and not the remediation configs. Will sort that out